misk/web/tabs/loader/src/sagas/loaderSaga.ts

```diff
@@ -0,0 +1,102 @@
+// In case you need to use a selector
+// import also select from redux-saga/effects
+// and then simplie yield select(yourSelector())
+//
+// In case you need to redirect to whatever route
+// import { push } from react-router-redux and then
+// yield put(push('/next-page'))
+
+import { IMiskAdminTab, IMiskAdminTabs } from "@misk/common"
+import axios from "axios"
+import { all, call, put, takeEvery, takeLatest } from "redux-saga/effects"
+
+import {
+  dispatchLoader, IAction, IActionType, LOADER
+} from "../actions"
+
+function * handleGetAllComponents (action: IAction<IActionType, { adminTabs: IMiskAdminTabs }>) {
+  try {
+    let adminTabs: any = {}
+    try {
+      const { data } = yield call(axios.get, "http://0.0.0.0:8080/api/admintab/all")
+      adminTabs = data.adminTabs
+      yield put(dispatchLoader.success({ adminTabs }))
+      if (Object.entries(adminTabs).length === 0) {
+        yield put(dispatchLoader.success({ adminTabComponents: { } }))
+      }
+    } catch (e) {
```

Comment:
Note: Nested try/catch that are doing the same thing are a clear sign there's something to be re-thought. As discussed in person I think the try/catch should be very targeted. If these functions were required to never throw errors then the function that calls them would likely protect against it.

Reply:
+1, refactored with the principles we discussed offline